server/src/main/java/org/apache/druid/segment/metadata/AbstractSegmentMetadataCache.java

```diff
@@ -805,6 +808,11 @@ public Set<SegmentId> refreshSegmentsForDataSource(final String dataSource, fina
     return retVal;
   }
 
+  void logSegmentsToRefresh(String dataSource, Set<SegmentId> ids)
```

Comment:
Could you add java docs to this method. Also there should be a abstract method called logRefreshSegments and that should return a boolean. 
Then each impl can set true or false 
and then CSMC can return true where as the broker one can return false.

Reply:
What is the benefit of doing that? Do we need to utilise the return type? Also, is there a reason behind making it abstract?

Reply:
So the main logic of logging does not leak out to impls. 
So a rouge impl cannot mess up the logs of the coordinator ever . Atleast that is what I was thinking.

Reply:
Okay, but is there a reason behind making it abstract and implementing it in both Broker and Coordinator? 
The base method could return false and the overridden method in Coordinator could return true?

Reply:
I feel that the implementation logic for logging the segments should reside in the child classes. This gives flexibility to both Broker and Coordinator to log different number of segments, blacklist/whitelist datasources etc.

Reply:
> The base method could return false and the overridden method in Coordinator could return true?
Sure.

Reply:
> >The base method could return false and the overridden method in Coordinator could return true?

> Sure.

I haven't made this change yet. Let me know what do you think about this, 

>I feel that the implementation logic for logging the segments should reside in the child classes. This gives flexibility to both Broker and Coordinator to log different number of segments, blacklist/whitelist datasources etc.